init-script

```diff
@@ -135,20 +135,44 @@ bootsplash() {
     fi
 }
 
+create_by_name_mountpoints()
+{
+    mkdir -p /dev/block/bootdevice/by-name/
+    for dev in $@;
+    do
+        DEVICE=$(echo $dev | sed "s|/dev/||")
+        UEVENT=$(cat /sys/class/block/${DEVICE}/uevent)
+        PARTNAME=$(echo "$UEVENT" | grep "PARTNAME=" | sed "s/PARTNAME=//")
+        MAJOR=$(echo "$UEVENT" | grep "MAJOR=" | sed "s/MAJOR=//")
+        MINOR=$(echo "$UEVENT" | grep "MINOR=" | sed "s/MINOR=//")
+
+        # create a block device
+        mknod /dev/block/bootdevice/by-name/$PARTNAME b $MAJOR $MINOR
+    done
+}
 
 mount_stowaways() {
+    echo "########################## creating block/bootdevice/by-name mountpoints"
+    if [ -e /dev/mmcblk? ]
+    then
+        create_by_name_mountpoints /dev/mmcblk?p*
+    elif [ -e /dev/sd? ]
+    then
+        create_by_name_mountpoints /dev/sd??*
+    fi
```

Comment:
I just noticed, that this won't work on devices with a sdX internal storage and have a sdcard, because it will find the sdcard first and then ignore the sdX partitions